### ISSUE TITLE
nvd3 tooltips not coming on mouse over at the same place in internet explorer in Modal window

### ISSUE THREAD

**Issue by NSudhir** [novus/nvd3#966]
Hello All,
We are using NVD3 and D3 java script libraries and we are integrating this with salesforce analaytics api. The way which we want these charts to be used is we do have different landing pages(VF Pages )in our org and there is a custom button on that Page. So on click of that button i want to display charts in a modal ( Namely Multibar, Discrete and Horizontal charts). I included those charts in vf page and the tooltips for those charts are very wierd coming at different places in the modal. So we decided to place each chart in the iframe so i think that solved my issue. The big issue which we are in to right now is the browser compatibility. The charts are not same in Ie9 as what they are looking in chrome.
So any suggestions would be a great help.Thank you in advance:)

**Comment by liquidpele** [comment]
We refactored tooltips in the current development branch, which is also in the 1.8.1-alpha right now... that will likely fix your issues, let us know! 

https://github.com/novus/nvd3/tree/v1.8.1-alpha

**Comment by NSudhir** [comment]
Sure will check it and let you know. Thank you.

**Comment by NSudhir** [comment]
Hi that completely knocked out my tooltips. I am not able to see 1 either!!

**Comment by liquidpele** [comment]
are you perhaps using the tooltipContent option?  that part changed some and you will need it to redo the custom tooltip if that's the case....  just look at the examples from that branch to see how they're doing it now but the gist is that you should use chart.tooltip.contentGenerater

**Comment by liquidpele** [comment]
for is this an IE 9 thing?

**Comment by NSudhir** [comment]
Yes ppreviously when i am refering 1.1.15b version the tool tips are behaving in the way i want in chrome but they are moving to far right in IE 9. 
I just saw a glimse of tooltip which is to far left after uploading the 1.8 version css and java script!

**Comment by NSudhir** [comment]
And right now we are displaying the charts in the modal window. Previously we used to display it on the visual force page. in that i write a custom tooltip code based on pixels of the page . i will paste the custom tooltip  code for one of the chart. But it seems like that javascript is not applying on the modal window which makes my job hard.


$(document).ready(function () {


   $("#LineChart").mouseover(function(e) {
                $("#CustomTooltip").removeClass('.nv-pointer-events-none');
             alert('hiiii');
              var style = document.createElement('style');
               style.type = 'text/css';
                style.innerHTML = '.cssClass {top: '+e.pageX+'px ; left: '+e.pageX+'px !important; opacity: 1; position: obsolute; }';
                document.getElementsByTagName('head')[0].appendChild(style);
            
       });

**Comment by liquidpele** [comment]
Yea, I haven't tested with IE9 in a while...  if you figure out what's wrong with it you can send us a pull request with the fix, but I really only worry about IE10+ for myself... 

I can't really tell what you're doing with that tooltip code... 

http://nvd3-community.github.io/nvd3/examples/site.html

It should have examples for charts with custom tooltips I think... or check out the tooltip object documentation: 

http://nvd3-community.github.io/nvd3/examples/documentation.html#tooltip   (specifically check out the contentGenerator option there)

You really should't have to use jquery at all...

**Comment by NSudhir** [comment]
Let me know one thing pls. how to set the custom tool tip!!
And the second question is will the chart tooltips aligns correctly in Modal window. Because i am using mulitiple charts in the modal window (Multi, Horizontal, Discrete). So as i said before the chart tooltips were all over the window.So i really dont know how to approach this issue, so i come up with iframes for each chart and now the chart tooltips are working fine in chrome but not in IE9,11. So not sure how to fix the browser compatibility issue now?

**Comment by NSudhir** [comment]
![novus1capture](https://cloud.githubusercontent.com/assets/10520800/7204236/8ecfffc4-e4ee-11e4-8658-2c4834999809.PNG)
This is how it looks in IE9 i am placing my cursor at the blue dot in pic and i am getting the tooltip at the pointer. This is working fine in chrome.
Any suggestions on that! if you want me to place the code i am using i can do that. Pls let me know.
Thank you.

**Comment by liquidpele** [comment]
That doesn't look like the new default tooltip format...   you should probably try removing all your custom tooltip stuff and see if it works, then add back formatting from that point.

**Comment by NSudhir** [comment]
Right now i dont have any custom tolltip file referencing in the code.

**Comment by NSudhir** [comment]
Is there a way to paste the code here!! So that you can see what exactly i am doing. I am using bootstrap columns too.

**Comment by liquidpele** [comment]
Then it doesn't look like you're using the latest nvd3 cod3 from https://github.com/novus/nvd3/tree/v1.8.1-alpha

You can create a jsfiddle or plunkr to demo it if you want.

**Comment by NSudhir** [comment]
Yh i tried referencing the latest nvd3 and referencing that in component, but as i said before when i try doing that the the tool-tips are completely disappearing.

**Comment by liquidpele** [comment]
Well, the old tooltips had known bugs with positioning, so you'll need to try the new ones and debug the issues with the new ones...

**Comment by liquidpele** [comment]
If you can create a jsfiddle or plunkr to demonstrate the issue with the new code then I can take a look for you, but it's not worth it to debug the older stuff since it was already deleted/refactored.

**Comment by NSudhir** [comment]
Oh ok. i will try again referencing the new js library and paste the pic of how tooltip looks like.Thank you.

**Comment by liquidpele** [comment]
A pic isn't going to help much I don't think, I'll need to see a demo so I can debug it with dev tools.

**Comment by NSudhir** [comment]
Sure.

**Comment by NSudhir** [comment]
is there any updated version for d3 javascript library after this 3.5.3? right now i am referencing 3.5.3!

**Comment by liquidpele** [comment]
Any 3.5.x version is fine, that's what I use as well.

**Comment by NSudhir** [comment]
ok.

**Comment by NSudhir** [comment]
Hi after uploading that new js library and referencing i got the tool tips finer than with the previous version but still in IE9 they stick at a single place. the attached is the image. 
![ienovuscapture](https://cloud.githubusercontent.com/assets/10520800/7210302/b99e63dc-e51e-11e4-9bec-3a13ea6b64f5.PNG)

**Comment by NSudhir** [comment]
i never used both of the jsfiddle and plunker before. so i dont know how they work. So can i paste my code here?

**Comment by liquidpele** [comment]
https://jsfiddle.net/ - you can create example pages there to share, it's a very widely used site so it'll be good to learn how to use.   Please do not paste your code in the discussion.

**Comment by NSudhir** [comment]
Seems like its pretty hard to paste the entire code because i have that entire code in between sales force tags  and i am using these charts as a part of the pages. The individual charts works absolutely fine and no issues with the tooltips either. When i try to combine all these charts the issue of tooltip arising which solves by using iframe. 
The suggestion i would like to hear is did you heard any tooltips issues in IE9, because even while combining all the charts and rendering in Modal the tooltips are working fine in chrome.
But they are fixing to the left most of the page in IE9. 
So any help on this issue would be a great help.

**Comment by NSudhir** [comment]
And i am taking the data from the report and displaying it in a chart

**Comment by liquidpele** [comment]
So is it working correctly in everything except IE9?

**Comment by NSudhir** [comment]
yes.tooltips are not correct in ie

**Comment by NSudhir** [comment]
![casecapture](https://cloud.githubusercontent.com/assets/10520800/7222106/83862ad4-e6d4-11e4-884b-c4994657783c.PNG)
Hi this is how tooltip looks in ie9.
my cursor is at red arrow and tooltip is at blue arrow. Tooltip position remains constant where ever i place my cursor on the chart. Position is pretty much fixed in IE. 
Can you help with this issue please.Thank you.

**Comment by liquidpele** [comment]
I can't even install IE9 anymore... the only download for it is for Vista  :/

**Comment by NSudhir** [comment]
Oh ok. So any quick suggestion on that!!

**Comment by joshjordan** [comment]
@liquidpele https://www.modern.ie/en-us/virtualization-tools

**Comment by NSudhir** [comment]
Hi Josh thank you for responding. how am i going to approach this?

**Comment by NSudhir** [comment]
oh sorry.i thought it was for me:)

**Comment by NSudhir** [comment]
@liquidpele Hi did you got an chance to try them in ie9?

**Comment by liquidpele** [comment]
I won't have the time for a while and IE9 just isn't a big priority for me.  If you wan to try and work on it and send any fixes via a pull request we'll look though :)

**Comment by NSudhir** [comment]
OK. Thank you.
